test/export/test_export.py

```diff
@@ -737,6 +738,61 @@ def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
         self.assertEqual(gm(*args), m(*args))
         args = (torch.randn(15, 3, 256, 256), torch.ones(15, 32, 256, 256))
         self.assertEqual(gm(*args), m(*args))
+    
+    @testing.expectedFailureLegacyExportNonStrict  # Some small change due to unbacked values getting regenerated
+    @testing.expectedFailureLegacyExportStrict  # Some small change due to unbacked values getting regenerated
+    def test_export_max_nonstrict(self):
+        class FooMax(torch.nn.Module):
+            def forward(self, x):
+                return torch.ones(max(x.item(), 1024))
+        
+        ep_strict_foo_max_symint = export(FooMax(), (torch.tensor(4),), strict=True).graph
+        ep_non_strict_foo_max_symint = export(FooMax(), (torch.tensor(4),), strict=False).graph
+    
+        self.assertEqual(str(ep_strict_foo_max_symint), str(ep_non_strict_foo_max_symint))
```

Comment:
This is an interesting way to test compatibility...calling export multiple times in each variation, which means a bit of duplication across tests. Couldn't you just check results are as expected? It's not that we were generating incorrect results before, we weren't even passing export. So results should be enough.

Another thing to test is to compare with `sym_max` and `sym_min`, since that was the workaround. Would the graph be the same?

Reply:
The high level ask here is to do self-contained testing that everybody can pass independently instead of cross-mode testing.